akka-stream-tests/src/test/scala/akka/stream/scaladsl/HubSpec.scala

```diff
@@ -370,37 +371,43 @@ class HubSpec extends StreamSpec {
   "PartitionHub" must {
```

Comment:
Do we also need test coverage for:
* when partition function throws
* when materializer is abruptly terminated? 
* subscribers could actually even be materialized with separate materialisers from the one that materialized the hub, not sure if that is important to test as well?